test/lib/mayaUsd/nodes/testProxyShapeBase.py

```diff
@@ -44,6 +48,134 @@ def testBoundingBox(self):
+        self.assertEqual(1, len(ufe.Hierarchy.hierarchy(proxyShapeItem).children()))
+
+    def testDuplicateProxyStageFileBacked(self):
+        '''
+        Verify stage from file is duplicated properly.
+        '''
+        # open tree.ma scene
+        mayaUtils.openTreeScene()
+
+        # clear selection to start off
+        cmds.select(clear=True)
+
+        # select a USD object.
+        mayaPathSegment = mayaUtils.createUfePathSegment('|Tree_usd|Tree_usdShape')
+        usdPathSegment = usdUtils.createUfePathSegment('/TreeBase')
+        treebasePath = ufe.Path([mayaPathSegment])
+        treebaseItem = ufe.Hierarchy.createItem(treebasePath)
+
+        # TreeBase has two children
+        self.assertEqual(1, len(ufe.Hierarchy.hierarchy(treebaseItem).children()))
+
+        # get the USD stage
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # by default edit target is set to the Rootlayer.
+        self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetRootLayer())
+
+        # add treebaseItem to selection list
+        ufe.GlobalSelection.get().append(treebaseItem)
+
+        # duplicate 
+        cmds.duplicate()
+
+        # get the next item in UFE selection list.
+        snIter = iter(ufe.GlobalSelection.get())
+        duplStageItem = next(snIter)
+
+        # duplicated stage name is expected to be incremented correctly
+        self.assertEqual(str(duplStageItem.path().back()), "Tree_usd1")
+
+        # duplicated proxyShape name is expected to be incremented correctly
+        duplProxyShapeHier = ufe.Hierarchy.hierarchy(duplStageItem)
+        duplProxyShapeItem = duplProxyShapeHier.children()[0]
+        self.assertEqual(str(duplProxyShapeItem.path().back()), "Tree_usd1Shape")
+
+        # duplicated ProxyShapeItem should have exactly one children
+        self.assertEqual(1, len(ufe.Hierarchy.hierarchy(duplProxyShapeItem).children()))
+
+        # child name is expected to be Capsule1
+        childName = str(ufe.Hierarchy.hierarchy(duplProxyShapeItem).children()[0].path().back())
+        self.assertEqual(childName, "TreeBase")
+
+        # validate session name and anonymous tag name 
+        duplStage = mayaUsd.ufe.getStage(str(ufe.PathString.path('|Tree_usd1|Tree_usd1Shape')))
+        self.assertEqual(duplStage.GetLayerStack()[0], duplStage.GetSessionLayer())
+        self.assertEqual(duplStage.GetEditTarget().GetLayer(), duplStage.GetRootLayer())
+        self.assertEqual(True, "-session" in duplStage.GetSessionLayer().identifier)
+        self.assertEqual(True, "tree.usd" in duplStage.GetRootLayer().identifier)
+
+        # delete "/TreeBase"
+        cmds.delete('|Tree_usd1|Tree_usd1Shape,/TreeBase')
+
+        # confirm that edits are shared. Both source and duplicated proxyShapes have no children now.
+        self.assertEqual(0, len(ufe.Hierarchy.hierarchy(duplProxyShapeItem).children()))
+        self.assertEqual(0, len(ufe.Hierarchy.hierarchy(treebaseItem).children()))
 
```

Comment:
Added new tests cases to back up the logics in ProxyShapeBase.

Reply:
Thanks for the test!